### PR TITLE
Add support for apiToken and includeDetails flag on queries

### PR DESCRIPTION
**Pull request by andreas-marschke**
## CLI
Adds `-T, --apitoken <apitoken>` flag to provide an optional apiToken instead of username/password

Adds `-d, --details <boolean>` flag to enable/disable retrieving a detailed response from the server for Objects when querying

## API
Adds `connectByApiToken(tenantName, apiToken, callback)` to use a apiToken and optional tenant as authentication method

Extends `queryObjects(type, query, includeDetails, callback)` supporting includeDetails as flag to the query function. If not provided will use 3rd param as callback if it's a function

### CODE REVIEW

File: cli/cmd.js [hunk 1]
```diff
@@ -18,17 +18,19 @@ program
...
     .option("-u, --username <username>", "User name")
     .option("-p, --password <password>", "Password")
+    .option("-T, --apitoken <apitoken>", "SSO API Token")
     .option("-t, --tenant <tenant>", "Tenant name")
     .option("-r, --repository <url>", "Repository URL")
     .option("-j, --json", "Output as JSON")
     .option("-o, --output <file>", "Output file")
     .option("-v, --verbose", "Verbose debugging")
     .option("-a, --auth <file>", "auth.json file containing credentials")
+    .option("-d, --details <boolean>", "Enable/Disable retrieving full objet details")
```
Comment (nicjansma) [inline cli/cmd.js]: s/objet/object/

File: cli/core.js [hunk 2]
```diff
@@ -58,16 +58,18 @@ exports.init = function(options) {
         options.parent.repository = constants.REPOSITORY_URL;
     }
 
+  if (!options.parent.apiToken) {
     // ensure username and password have been specified
     if (!options.parent.username) {
-        log.error("--username (-u) is required");
-        process.exit(1);
+      log.error("--username (-u) is required");
```
Comment (nicjansma) [inline cli/core.js]: These were changed from 4-space to 2-space?

File: cli/core.js [hunk 3]
```diff
@@ -77,10 +79,17 @@ exports.init = function(options) {
  * @param {function(err, repo)} callback Callback
  */
 exports.connectToRepository = function(options, callback) {
-    var repo = new SOASTA.Repository(options.parent.repository);
+  var repo = new SOASTA.Repository(options.parent.repository);
```
Comment (nicjansma) [inline cli/core.js]: These should be 4-spaced

File: cli/core.js [hunk 4]
```diff
@@ -77,10 +79,17 @@ exports.init = function(options) {
...
+  if (options.parent.apiToken) {
+    repo.connectByApiToken(options.parent.tenantName || null, options.parent.apiToken, function(err) {
+      return callback && callback(err, repo);
+    });
+  }
+  else {
     repo.connect(options.parent.tenant, options.parent.username, options.parent.password, function(err) {
-        return callback && callback(err, repo);
+      return callback && callback(err, repo);
```
Comment (nicjansma) [inline cli/core.js]: These should be 4-spaced

Comment (andreas-marschke) [inline cli/core.js]: @nicjansma Fixed

File: cli/core.js (hunk 3 above)
Comment (andreas-marschke) [inline cli/core.js]: @nicjansma Fixed

**Pull request merged**